Remove address from Sign method on ChainSigner

Ayiga opened the pull request:
<!-- Closes #<ISSUE_NUMBER> -->
<!-- These comments should help create a useful PR message, please delete any remaining comments before opening the PR. -->
<!-- If there is no issue number make sure to describe clearly *why* this PR is necessary. -->
<!-- Mention open questions, remaining TODOs, if any -->

### This PR:
<!-- Describe what this PR adds to this repo and why -->
<!-- E.g. -->
<!-- * Implements feature 1 -->
<!-- * Fixes bug 3 -->
Removes the `Address` field from the `Sign` method on `ChainSigner`.  The address being passed to the `Sign` call in the batcher/driver.go file was likely incorrect.  By removing the address, we can ensure that we can still sign without requiring direct access to the signing key.

Additionally, this adds tests to illustrate that the Sign method works, and can be verified.

<!-- ### This PR does not: -->
<!-- Describe what is out of scope for this PR, if applicable. Leave this section blank if it's not applicable -->
<!-- This section helps avoid the reviewer having to needlessly point out missing parts -->
<!-- * Implement feature 3 because that feature is blocked by Issue 4 -->
<!-- * Implement xyz because that is tracked in issue #123. -->
<!-- * Address xzy for which I opened issue #456 -->

<!-- ### Key places to review: -->
<!-- Describe key places for reviewers to pay close attention to -->
<!-- * file.rs, `add_integers` function -->
<!-- Or directly comment on those files/lines to make it easier for the reviewers -->

<!-- ### How to test this PR:  -->
<!-- Optional, uncomment the above line if this is relevant to your PR -->
<!-- If your PR is fully tested through CI there is no need to add this section -->
<!-- * E.g. `just test` -->

<!-- ### Things tested -->
<!-- Anything that was manually tested (that is not tested in CI). -->
<!-- E.g. building/running of docker containers. Changes to docker demo, ... -->
<!-- Especially mention anything untested, with reasoning and link an issue to resolve this. -->

<!-- Complete the following items before creating this PR -->
<!-- [ ] Issue linked or PR description mentions why this change is necessary. -->
<!-- [ ] PR description is clear enough for reviewers. -->
<!-- [ ] Documentation for changes (additions) has been updated (added).  -->
<!-- [ ] If this is a draft it is marked as "draft".  -->

<!-- To make changes to this template edit https://github.com/EspressoSystems/.github/blob/main/PULL_REQUEST_TEMPLATE.md -->


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1209743774312205

Ayiga committed:
Remove address from Sign method on ChainSigner

The address being passed to the ChainSigner's Sign function is likely the incorrect
address.  Additionally, the address is already possible to be known already and is
therefore unnecessary to pass in.

The change removes the need to pass in the address of the key that is being used to
sign the transactions.

Additionally, tests have been added to verify that signature verification works as expected.

dailinsubjam commented on op-service/crypto/espresso_test.go:
```diff
@@ -0,0 +1,83 @@
+
+	var pubKey common.Address
+	copy(pubKey[:], pubKeySlice)
+	require.Equal(t, expectedAddr, addr)
```
This line is same as line 68 right? Maybe what we want here is `require.Equal(t, expectedAddr, pubKey)`.

Ayiga replied:
Nice catch.

Changing this actually makes the test fail as well currently.

Ayiga replied:
Fixed in 647f5e5188184d869848ae334bfe7784e8c7548e.

It has also been annotated with comments that illustrate how to successfully derive an `Address` from `Signature` utilizing the recovered public key.

dailinsubjam replied:
Hooray! This one works!

Ayiga committed:
Fix public address derivation in `testChainSignerSign`

The equality check on espresso_test.go:81 was a copy of espresso_test.go:68
as correctly pointed out by @dalinsubjam.  This is not what we want.  Instead we
want to be able to derive the address from the public key that is recovered from
the signature, and to test that against our expected address.

This modifies the logic to manually derive the address from the recovered public
key utilizing publicly available functions within the go-ethereum repository. As
a result, the test has been corrected, and now passes successfully.
